clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1307,6 +1404,53 @@ raw2trace_t::process_tasks(std::vector<raw2trace_thread_data_t *> *tasks)
     }
 }
 
+std::string
+raw2trace_t::read_syscall_template_file()
+{
+    if (syscall_template_file_ == nullptr)
+        return "";
+    trace_entry_t entry;
+    int last_syscall_num = -1;
+    bool first_entry_for_syscall = false;
+    while (syscall_template_file_->read((char *)&entry, sizeof(entry))) {
+        // Track encodings for system call template instructions. We do not need the
+        // returned entry memref count, but only the encoding locations that we will
+        // query using get_decode_pc later.
+        syscall_trace_template_encodings_.entry_memref_count(&entry);
+        if (entry.type == TRACE_TYPE_MARKER) {
+            switch (entry.size) {
+            case TRACE_MARKER_TYPE_SYSCALL:
+                last_syscall_num = static_cast<int>(entry.addr);
+                first_entry_for_syscall = true;
+                continue;
+            case TRACE_MARKER_TYPE_FILETYPE:
+                // We cannot at this point verify that the trace being post-processed is
+                // of the same arch. We do that later in write_syscall_template.
+                syscall_template_file_type_ =
+                    static_cast<offline_file_type_t>(entry.addr);
+                if (!TESTANY(OFFLINE_FILE_TYPE_KERNEL_SYSCALL_TRACE_TEMPLATES,
+                             entry.addr))
+                    return "Invalid file type for system call trace template file.";
+                if (!TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, entry.addr)) {
+                    return "System call trace templates without encodings are not "
+                           "supported.";
+                }
+                continue;
+            }
+        }
+        if (entry.type == TRACE_TYPE_THREAD_EXIT)
```

Comment:
Should we validate other entry types too, or we're ok with whatever other records are in there?

Reply:
For now let's start with being okay with whatever the template file provides.